File: caregiver_dashboard/backend/models/medicationSchedule.js
module.exports = (sequelize, DataTypes) => {
    const MedicationSchedule = sequelize.define('MedicationSchedule', {
      ScheduleID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      ScheduledTime: DataTypes.TIME,
      PatientMedicationID: DataTypes.INTEGER,
    }, {
      tableName: 'medicationschedules',
      timestamps: false,
    });
  
    MedicationSchedule.associate = function(models) {
      MedicationSchedule.belongsTo(models.PatientMedication, { foreignKey: 'PatientMedicationID' });
    };
  
    return MedicationSchedule;
  };
  

File: caregiver_dashboard/frontend/lib/models/medication_schedule.dart
class MedicationSchedule {
  int scheduleId;
  String scheduledTime;
  int patientMedicationId;

  MedicationSchedule({
    required this.scheduleId,
    required this.scheduledTime,
    required this.patientMedicationId,
  });

  factory MedicationSchedule.fromJson(Map<String, dynamic> json) {
    return MedicationSchedule(
      scheduleId: json['ScheduleID'],
      scheduledTime: json['ScheduledTime'],
      patientMedicationId: json['PatientMedicationID'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ScheduleID': scheduleId,
      'ScheduledTime': scheduledTime,
      'PatientMedicationID': patientMedicationId,
    };
  }
}
